src/server/req_delete.c

```diff
@@ -235,6 +235,9 @@ check_deletehistoryjob(struct batch_request * preq)
 		}
 	}
 
+	if ((histpjob = find_job(jid)) == NULL)
+		return FALSE;
+
```

Comment:
It looks like the above block of code is not required. Just do a test case and verify if it is indeed needed.

Reply:
Fixed in rc 33 https://github.com/PBSPro/pbspro/pull/590/commits/3da33d42343d89615cf55f6531ed5362ebcdd287 and verified.